lib/lib-dynamodb-data-mapper/src/marshaller/DataMarshaller.ts

```diff
@@ -0,0 +1,73 @@
+import type { AttributeValue } from "@aws-sdk/client-dynamodb";
+import { marshallOptions } from "@aws-sdk/util-dynamodb";
+
+import { ItemSchema } from "../schema";
+import { marshallItem, marshallKey } from "./marshall";
+import { unmarshallItem, unmarshallPlainObject } from "./unmarshall";
+
+/**
+ * Schema-aware marshalling and unmarshalling utilities.
+ *
+ * This class provides static methods to transform JavaScript objects
+ * to and from DynamoDB AttributeValue format using schema definitions.
+ */
+export class DataMarshaller {
```

Comment:
Why do we need this abstraction, seems like we already have all methods under marshal/